apps/full-site-editing/full-site-editing-plugin/editor-site-launch/src/launch-steps/name-step/index.tsx

```diff
@@ -16,29 +17,56 @@ import LaunchStepContainer, { Props as LaunchStepProps } from '../../launch-step
 import './styles.scss';
 
 const NameStep: React.FunctionComponent< LaunchStepProps > = ( { onNextStep } ) => {
-	const { setStepComplete } = useDispatch( LAUNCH_STORE );
+	const domain = useSelect( ( select ) => select( LAUNCH_STORE ).getSelectedDomain() );
+	const { setStepComplete, setStepIncomplete, setDomainSearch } = useDispatch( LAUNCH_STORE );
+	const [ title, setTitle ] = useEntityProp( 'root', 'site', 'title' );
+	const { saveEditedEntityRecord } = useDispatch( 'core' );
 
-	const handleClick = () => {
-		setStepComplete( LaunchStep.Name );
+	const handleSave = () => {
+		saveEditedEntityRecord( 'root', 'site' );
+
+		// update domainSearch only if there is no custom Domain selected
+		! domain && setDomainSearch( title );
+
+		title.length ? setStepComplete( LaunchStep.Name ) : setStepIncomplete( LaunchStep.Name );
+	};
+
+	const handleNext = () => {
+		handleSave();
 		onNextStep?.();
 	};
 
 	return (
 		<LaunchStepContainer className="nux-launch-name-step">
 			<div className="nux-launch-step__header">
 				<div>
-					<Title>{ X__( 'Name your site', 'full-site-editing' ) }</Title>
-					<SubTitle>{ X__( 'Pick a name for your site.', 'full-site-editing' ) }</SubTitle>
+					<Title>{ __( 'Name your site', 'full-site-editing' ) }</Title>
+					<SubTitle>{ __( 'Pick a name for your site.', 'full-site-editing' ) }</SubTitle>
 				</div>
+				<ActionButtons>
+					<NextButton onClick={ handleNext } disabled={ ! title } />
+				</ActionButtons>
 			</div>
 			<div className="nux-launch-step__body">
-				<p>
-					For now, this page serves as a demo of how to mark a step as complete and go to the next
-					step.
-				</p>
-				<Button onClick={ handleClick } isPrimary>
-					Mark Complete &amp; Go To Next Step
-				</Button>
+				<form onSubmit={ handleNext }>
+					<TextControl
+						id="nux-launch-step__input"
+						className="nux-launch-step__input"
+						onChange={ setTitle }
+						onBlur={ handleSave }
+						value={ title }
+						spellCheck={ false }
+						autoComplete="off"
+						placeholder={ __( 'Enter site name', 'full-site-editing' ) }
+						autoCorrect="off"
+						data-hj-whitelist
+					/>
+					<p className="nux-launch-step__input-hint">
+						<Tip size={ 18 } />
+						{ /* translators: The "it" here refers to the site title. */ }
+						<span>{ __( "Don't worry, you can change it later.", 'full-site-editing' ) }</span>
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **30** times:
`translate( 'Don't worry, you can change it later.' )` *ES Score: 12*